### PR TITLE
Wording, syntax, spelling, grammar fixes

### PR DESCRIPTION
**Pull request by LindsayHill**
Working my way through cleaning up language, spelling, grammar, fixing RST syntax issues, alignment, random old `st2_deploy.sh` references, etc.

Going to go through all content page by page...slow work...

### CODE REVIEW

File: docs/source/development/index.rst [hunk 1]
```diff
@@ -23,82 +23,77 @@ For an overview of core StackStorm code structure, please refer to
...
 
-You are encouraged to generously use the log statements across the code base -
-you should log every event which increases the visibility and / or makes the
-product easier to debug and support.
+You are encouraged to generously use the log statements across the code base. You should log every
+event which increases the visibility and/or makes the product easier to debug and support.
 
-Every log statement should also include as much as useful additional context as
-possible. This context should be included in the dictionary which is passed via
-``extra`` keyword argument to the logger method as shown bellow.
```
Comment (Kami) [inline docs/source/development/index.rst]: I'm sure all those `bellow` occurrences were me :trollface: 

Accidentally added `bellow` as a correct spelling to my vim dictionary, but I fixed that like a year ago ;P

Comment (LindsayHill) [reply docs/source/development/index.rst]: Ha! Every time I think I've got rid of them all, another one pops up. I think there's a few in the docstrings too

**Comment by LindsayHill** [comment]
This PR is starting to get a bit big. Working from the bottom of the docs menu up, I'll finish roadmap, upgrades, development, troubleshooting and reference sections, then look to merge. Then will continue working up through rest of docs as a separate PR

**Comment by LindsayHill** [comment]
OK, at 42 files changed this PR is probably big enough.

**Pull request merged**